sonar-text-plugin/src/main/java/org/sonar/plugins/common/TextFilePredicate.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarQube Text Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.common;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.sonar.api.batch.fs.FilePredicate;
+import org.sonar.api.batch.fs.InputFile;
+
+public class TextFilePredicate implements FilePredicate {
+
+  private final Set<String> extensions;
+
+  public TextFilePredicate(String... suffixes) {
+    extensions = Arrays.stream(suffixes)
+      .map(String::trim)
+      .filter(value -> !value.isEmpty())
+      .collect(Collectors.toSet());
+  }
+
+  @Override
+  public boolean apply(InputFile inputFile) {
+    String filename = inputFile.filename();
+    String extension = extension(filename);
```

Comment:
Shouldn't there already be a method to get the extension from the input File?

Reply:
Not sure what you mean here, `InputFile` provide a limited set of method, and nothing related to the extension itself, so we have to compute it ourselves.

Reply:
Let me have a look if I find something.

Reply:
Basically you don't need `TextFilePredicate`:

You could use something like this (but cleaned up, with more better collection handling.):
```
    List<FilePredicate> extensionPredicates = new ArrayList<>();
    for (String extension : extensions) {
      FilePredicate filePredicate = sensorContext.fileSystem().predicates().hasExtension(extension);
      extensionPredicates.add(filePredicate);
    }

    FilePredicate allExtensions = sensorContext.fileSystem().predicates().or(extensionPredicates);
```
This would replace your `textFilePredicate` in the Sensor.

Reply:
Good catch!